clients/cobol-lsp-vscode-extension/src/services/copybook/CopybookMessageHandler.ts

```diff
@@ -134,3 +149,36 @@ export function downloadCopybookHandler(
     quietMode,
   );
 }
+
+async function getE4ECopyBookLocation(
+  copybookName: string,
+  documentUri: string,
+  storagePath: string,
+) {
+  const config = await E4ECopybookService.getE4EClient(documentUri, undefined);
+  if (!config) {
+    throw Error;
+  }
+  const pgm = getProgramNameFromUri(documentUri);
+
+  const mapped_pgm = config.elements.find((x) => x.element === copybookName)
+    ?.use_map
+    ? path.join(USE_MAP, pgm)
+    : pgm;
```

Comment:
I still don't understand the qualification by program name only.

Reply:
it is defined with   `CopybookURI.createDatasetPath(
      profileAsString(config.profile),
      mapped_pgm,
      storagePath,
      E4E_FOLDER,
    )`
    including mapped, program name, and environment specs.

Reply:
Where are the "environment specs", I see only pgm and profile being used.